flytekit/bin/entrypoint.py

```diff
@@ -93,23 +98,40 @@ def _dispatch_execute(ctx: FlyteContext, task_def: PythonTask, inputs_path: str,
             output_file_dict = {_constants.FUTURES_FILE_NAME: outputs}
         else:
             _logging.getLogger().error(f"SystemError: received unknown outputs from task {outputs}")
-            # TODO This should probably cause an error file
-            return
-
-        for k, v in output_file_dict.items():
-            _common_utils.write_proto_to_file(v.to_flyte_idl(), _os.path.join(ctx.execution_state.engine_dir, k))
-
-        # Step3a
-        ctx.file_access.upload_directory(ctx.execution_state.engine_dir, output_prefix)
-        _logging.info(f"Outputs written successful the the output prefix {output_prefix}")
+            output_file_dict[_constants.ERROR_FILE_NAME] = _error_models.ErrorDocument(
+                _error_models.ContainerError(
+                    "UNKNOWN_OUTPUT",
+                    f"Type of output received not handled {type(outputs)} outputs: {outputs}",
+                    _error_models.ContainerError.Kind.RECOVERABLE,
+                )
+            )
+    except _scoped_exceptions.FlyteScopedException as e:
+        _logging.error("!! Begin Error Captured by Flyte !!")
+        output_file_dict[_constants.ERROR_FILE_NAME] = _error_models.ErrorDocument(
+            _error_models.ContainerError(e.error_code, e.verbose_message, e.kind)
+        )
+        _logging.error(e.verbose_message)
+        _logging.error("!! End Error Captured by Flyte !!")
     except Exception as e:
         if isinstance(e, IgnoreOutputs):
             # Step 3b
             _logging.warning(f"IgnoreOutputs received! Outputs.pb will not be uploaded. reason {e}")
             return
         # Step 3c
         _logging.error(f"Exception when executing task {task_def.name}, reason {str(e)}")
-        raise e
+        _logging.error("!! Begin Unknown System Error Captured by Flyte !!")
+        exc_str = _traceback.format_exc()
+        output_file_dict[_constants.ERROR_FILE_NAME] = _error_models.ErrorDocument(
```

Comment:
So I wanted to do this - in case of error (and this maybe because we are unable to write to S3/GCS etc) we should write the output to https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/#customizing-the-termination-message
This would make propeller read the outputs and write it to Flyteadmin.

Reply:
Can we backlog this as it's a bit different?

Reply:
We kinda do that... [Here](https://github.com/flyteorg/flyteplugins/blob/8a49ee2bbef9e89b6d1cec94b4eef1ee8fe7aca3/go/tasks/pluginmachinery/flytek8s/container_helper.go#L127)
and we do indeed take that message and bubble up to Admin (after shortening it)

Reply:
@EngHabu I know we do that in FlytePropeller, what I meant was do this in flytekit as well?

Reply:
Hmmm I'm not sure what do we NOT do currently (after this PR)? in the older flytekit, when we fail to write the outputs, we write THAT error into stdErr and that bubbles up... are you saying with this PR we don't yet do that? or are you saying "Also write the real outputs" to stdErr?

Reply:
no, i do not follow, where we write the error to stdErr